task_02/nikolai_chernyshov/main.cpp

```diff
@@ -0,0 +1,175 @@
+#include <memory>
+#include <cstdlib>
+#include <iostream>
+#include <fstream>
+
+struct edge {
+    static int edge_id_max;
+    int edge_id;
+    int to_vertice;
+    std::shared_ptr<struct vertice> ptr;
+
+    edge() {}
```

Comment:
Для внутренних `C++` оптимизаций, если хочешь иметь дефолтный конструктор, лучше записывать так:
```suggestion
    Edge() = default;
```